src/main/java/core/basesyntax/MatrixOperation.java

```diff
@@ -6,18 +6,49 @@
  */
 public class MatrixOperation {
     public int[] getDiagonal(int[][] matrix) {
-        return null;
+        int[] result = new int[matrix.length];
+        for (int i = 0; i < matrix.length; i++) {
+            result[i] = matrix[i][i];
+        }
+        return result;
     }
 
     public int[] getCounterDiagonal(int[][] matrix) {
-        return null;
+        if (matrix.length == 1) {
+            return matrix[0];
+        }
+
+        int[] result = new int[matrix[0].length];
+        int i = 0;
```

Comment:
good, but let's move the declaration of this variable in for loop too
Example:
https://www.geeksforgeeks.org/loop-java-important-points/
https://stackoverflow.com/a/14752913

Reply:
Done